### PR TITLE
Upgrade axum to 0.7.4

### PR DESCRIPTION
**Pull request by cscetbon**
Including in this PR:
- Upgrade of axum from 0.6.19 to 0.7.4
- Upgrade of dependencies that depend on axum
- Upgrade of env_logger crate
- Export axum from dapr::server::actor
- Use absolute paths in actor proc macro to use axum from dapr and not from the client
- Upgrade of toolchain as it was failing at resolving dependency/feature of env_logger

### CODE REVIEW

**Review by mikeee — COMMENTED** [review]
First look

File: src/server/actor/tests.rs
Comment (cscetbon) [reply src/server/actor/tests.rs:71]: Axum takes a `TcpListener` now and no longer a `SocketAddr`. It's just doing the same thing at both places, no need to do it differently and confuse anyone who reviews the code and wonder why one uses a way and another does it differently

File: src/server/utils.rs
Comment (cscetbon) [reply src/server/utils.rs:24]: Correct, see https://github.com/tokio-rs/axum/blob/axum-v0.7.4/axum/src/json.rs#L96-L99

Comment (cscetbon) [reply src/server/utils.rs:24]: It's also said in the changelog
```
The following types/traits are no longer generic over the request
body (i.e. the B type param has been removed):
  FromRequestParts
  FromRequest
  ...
```

**Comment by cscetbon** [comment]
@yaron2 Can you take a look ?

**Comment by yaron2** [comment]
> @yaron2 Can you take a look ?

LGTM, waiting on final approval from @mikeee.

FYI it would be extremely useful to add the ability to specify the Dapr endpoint to override 127.0.0.1 the same way other SDKs are doing through environment variables.

**Review by mikeee — APPROVED** [review]
LGTM! The upgrade path makes sense, thanks for the clarifications.

**Comment by cscetbon** [comment]
> FYI it would be extremely useful to add the ability to specify the Dapr endpoint to override 127.0.0.1 the same way other SDKs are doing through environment variables.

Let's do it in another PR, but it would need a good description cause I've read that _the dapr side car currently does not support invoking actors via grpc_, and other SDKs use an env variable called `DAPR_GRPC_ENDPOINT`.

That PR got a 👍 from both of you so it should be ready to merge 😉

**Pull request merged**
